src/app/(landing)/oauth-callback/apple/page.tsx

```diff
@@ -0,0 +1,84 @@
+"use client"
+
+import { Suspense, useEffect, useState } from "react"
+import { useRouter, useSearchParams } from "next/navigation"
+import { useAuth } from "@/providers/auth-provider"
+import { useTurnkey } from "@turnkey/sdk-react"
+import { Loader } from "lucide-react"
+
+import { Card, CardHeader, CardTitle } from "@/components/ui/card"
+import { Icons } from "@/components/icons"
+
+function OAuthProcessCallback() {
+  const searchParams = useSearchParams()
+
+  const { authIframeClient } = useTurnkey()
+  const { loginWithApple } = useAuth()
+  const router = useRouter()
+
+  const [storedToken, setStoredToken] = useState<string | null>(null) // Store the token locally
+  const [hasLoggedIn, setHasLoggedIn] = useState(false) // Track if loginWithOAuth has been called
+
+  // Get token from query string params and store in state when available
+  useEffect(() => {
+    const fragment = window.location.hash
+    if (fragment) {
+      const params = new URLSearchParams(fragment.slice(1)) // Remove the "#" and parse parameters
+      const token = params.get("id_token")
+      if (!token) {
+        const msg = "Invalid redirect parameters"
+        router.push(`/?error=${encodeURIComponent(msg)}`)
+        return
+      }
+      setStoredToken(token) // Store token if available
+    } else {
+      const msg = "Invalid redirect parameters"
+      router.push(`/?error=${encodeURIComponent(msg)}`)
+    }
+  }, [searchParams])
```

Comment:
I think since we are not using `searchParams` in this `useEffect` hook we can remove it as a dep and then we'll want to add the `router` as a dep of this hook since it is being used here
```suggestion
  }, [router])
```